### PR TITLE
Bug 1145280 - Create team landing pages

### PR DESCRIPTION
**Pull request by bobsilverberg**
This implements the team landing pages as discussed in bug 1145280.

@glogiotatidis | @akatsoulas r? please and thank you

### CODE REVIEW

File: oneanddone/tasks/filters.py [hunk 1]
```diff
@@ -36,6 +36,27 @@ class Meta:
...
 
 
+class SmallTasksFilterSet(django_filters.FilterSet):
+
+    difficulty = django_filters.ChoiceFilter(
+        choices=(
+            ('', 'Any'),
+            (Task.BEGINNER, 'Beginner'),
+            (Task.INTERMEDIATE, 'Intermediate'),
+            (Task.ADVANCED, 'Advanced')),
```
Comment (glogiotatidis) [inline oneanddone/tasks/filters.py]: Localization?

File: oneanddone/tasks/filters.py [hunk 2]
```diff
@@ -36,6 +36,27 @@ class Meta:
...
+        label=_lazy(u'Task Difficulty'),
+        required=False)
+
+    execution_time = django_filters.ChoiceFilter(
+        choices=(('', 'Any'), (15, 15), (30, 30), (45, 45), (60, 60)),
```
Comment (glogiotatidis) [inline oneanddone/tasks/filters.py]: Localizatoin?

File: oneanddone/tasks/filters.py [hunk 3]
```diff
@@ -43,6 +64,14 @@ class TasksFilterSet(django_filters.FilterSet):
         label=_lazy(u'Search for tasks')
     )
 
+    difficulty = django_filters.MultipleChoiceFilter(
+        choices=(
+            (Task.BEGINNER, 'Beginner'),
+            (Task.INTERMEDIATE, 'Intermediate'),
+            (Task.ADVANCED, 'Advanced')),
```
Comment (glogiotatidis) [inline oneanddone/tasks/filters.py]: Localization?

File: oneanddone/tasks/forms.py [hunk 4]
```diff
@@ -198,3 +199,31 @@ class TaskInvalidationCriterionForm(forms.Form):
 TaskInvalidCriteriaFormSet = forms.formsets.formset_factory(
     TaskInvalidationCriterionForm,
     formset=BaseTaskInvalidCriteriaFormSet)
+
+
+class TeamForm(forms.ModelForm):
+    url_code = forms.RegexField(
+        label=_("Team url suffix:"),
```
Comment (glogiotatidis) [inline oneanddone/tasks/forms.py]: You need `_lazy` for these

File: oneanddone/tasks/models.py [hunk 5]
```diff
@@ -312,11 +311,40 @@ def __unicode__(self):
...
+    def get_edit_url(self):
+        return reverse('tasks.edit_team', args=[self.id])
+
+    def get_friendly_url(self):
+        return reverse('tasks.team_short', args=[self.url_code])
+
+    @classmethod
+    def get_team_by_id_or_url_code(self, kwargs):
+        url_code = kwargs.get('url_code')
+        if url_code is None:
```
Comment (glogiotatidis) [inline oneanddone/tasks/models.py]: I'd use the more generic and simple `if not url_code`

File: oneanddone/tasks/models.py [hunk 6]
```diff
@@ -312,11 +311,40 @@ def __unicode__(self):
...
+            try:
+                return TaskTeam.objects.get(id=kwargs.get('pk'))
+            except ObjectDoesNotExist:
```
Comment (glogiotatidis) [inline oneanddone/tasks/models.py]: I've been bitten in the past while using the generic ObjectDoesNotExist exception and catching one that was raised for another object than the one I intended it to. This of course is a very simple try/except nevertheless I'd still go with TaskTeam.DoesNotExist

**Comment by glogiotatidis** [comment]
Just a few nits. Nice work! r+wc

File: oneanddone/tasks/models.py (hunk 6 above)
Comment (bobsilverberg) [inline oneanddone/tasks/models.py]: Thanks @glogiotatidis. I didn't even know about `TaskTeam.DoesNotExist` as a more specific version of `ObjectDoesNotExist`. Cool!

**Comment by bobsilverberg** [comment]
Thanks for the review @glogiotatidis! I'm still not used to thinking about l10n all the time, so I appreciate you finding those issues. I addressed all of your comments and will merge the PR.

**Pull request merged**